core/src/apps/cardano/sign_tx/__init__.py

```diff
@@ -1,30 +1,39 @@
-from typing import Type
-
-from trezor import log, messages, wire
-from trezor.enums import CardanoTxSigningMode
+from typing import TYPE_CHECKING
 
 from .. import seed
-from .signer import Signer
+
+if TYPE_CHECKING:
+    from typing import Type
+    from trezor.wire import Context
+    from trezor.messages import CardanoSignTxFinished, CardanoSignTxInit
 
 
 @seed.with_keychain
 async def sign_tx(
-    ctx: wire.Context, msg: messages.CardanoSignTxInit, keychain: seed.Keychain
-) -> messages.CardanoSignTxFinished:
+    ctx: Context, msg: CardanoSignTxInit, keychain: seed.Keychain
+) -> CardanoSignTxFinished:
+    from trezor.messages import CardanoSignTxFinished
+    from trezor import log, wire
+    from trezor.enums import CardanoTxSigningMode
+
+    from .signer import Signer
+
+    signing_mode = msg.signing_mode  # local_cache_attribute
```

Comment:
why is this not `signing_mode`?

edit: it appears that the simplification did not happen for most of Cardano